vault/resource_aws_secrets_sync_destination.go

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"context"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+)
+
+const (
+	fieldAccessKeyID     = "access_key_id"
+	fieldSecretAccessKey = "secret_access_key"
+
+	vaultFieldConnectionDetails = "connection_details"
+)
+
+var awsSyncDestinationFields = []string{
+	fieldAccessKeyID,
+	fieldSecretAccessKey,
+	consts.FieldAWSRegion,
+}
+
+var awsSyncNonSensitiveFields = []string{
+	consts.FieldAWSRegion,
+}
+
+func awsSecretsSyncDestinationResource() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: provider.MountCreateContextWrapper(awsSecretsSyncDestinationWrite, provider.VaultVersion115),
+		ReadContext:   provider.ReadContextWrapper(awsSecretsSyncDestinationRead),
+		DeleteContext: awsSecretsSyncDestinationDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			consts.FieldName: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Unique name of the AWS destination.",
+				ForceNew:    true,
+			},
+			fieldAccessKeyID: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Access key id to authenticate against the AWS secrets manager.",
+				ForceNew:    true,
+			},
+			fieldSecretAccessKey: {
+				Type:      schema.TypeString,
+				Optional:  true,
+				Sensitive: true,
+				Description: "Secret access key to authenticate against the AWS secrets " +
+					"manager.",
```

Comment:
Should we mention here as part of the description which env var can be used instead for each field on each destination?

Reply:
I added the environment variable information to the user/public facing documentation. Opted to keep the description strings within the internal resource itself short